### PR TITLE
feat: add TOC for supernested docs

### PR DESCRIPTION
**Pull request by simgobb**
Fixes [#4195](https://github.com/GoogleChrome/developer.chrome.com/issues/4195)

Changes proposed in this pull request:

- Add support for 4th level of nested TOC Docs
- Add FLEDGE API docs in TOC

### CODE REVIEW

**Comment by devnook** [comment]
@simgobb can you please provide url at which this change is visible/testable?

**Comment by simgobb** [comment]
@devnook yep. I've added some lower-tier pages to this page following the example in the issue description:

https://deploy-preview-4947--developer-chrome-com.netlify.app/docs/privacy-sandbox/overview/


On Device Auction for custom audiences -> FLEDGE API Developer Guide

The items nested there have been styled to support the extra nesting level.

File: site/_includes/macros/navigation-tree-sections.njk [hunk 1]
```diff
@@ -20,8 +21,8 @@ and marks it as active.
       </div>
       <span class="section-title">{{ item.title | i18n }}</span>
     </button>
-    <div class="navigation-tree__nested">
-      {{ navigationTreeSectionsInternal(item.sections) }}
+    <div class="navigation-tree__nested" depth={{tocDepth}}>
```
Comment (devnook) [inline site/_includes/macros/navigation-tree-sections.njk:24]: Please use data-* format for custom attributes.

**Comment by devnook** [comment]
@rachelandrew can you please PTAL for content TOC changes?

**Comment by heyawhite** [comment]
image

The style of this is a bit confusing. With the removed indentation and lack of any other differentiator, it's not clear that "Interest-based advertising" and "On-device custom auctions..." are associated with "Show relevant content.

Would it be possible to stylize it differently, _or_, per the original bug, on initial visitation to one of those inner pages, hide the extended outer menu? The top button "See full navigation" could reveal the full navigation.

I mocked up what this may look like 
image

**Comment by matthiasrohmer** [comment]
@heyawhite, we did improve the styling again. Could you take another look?

**Currently live**
image

**With this PR**
image

For now we refrained from your other proposal as it would cause quite a rework of the TOC stuff, so would this quick fix help you for now once we get more capacity?

**Comment by heyawhite** [comment]
> For now we refrained from your other proposal as it would cause quite a rework of the TOC stuff, so would this quick fix help you for now once we get more capacity?

@matthiasrohmer yep, this will work for now.

**Review by matthiasrohmer — APPROVED** [review]
Thanks all!

**Pull request merged**